internal/controllers/cluster/cluster_controller_status_test.go

```diff
@@ -0,0 +1,2336 @@
+		cluster                 *clusterv1.Cluster
+		reconcileDeleteExecuted bool
+		deletingReason          string
+		deletingMessage         string
+		expectCondition         metav1.Condition
+	}{
+		{
+			name:                    "deletionTimestamp not set",
+			cluster:                 fakeCluster("c"),
+			reconcileDeleteExecuted: false,
+			deletingReason:          "",
+			deletingMessage:         "",
+			expectCondition: metav1.Condition{
+				Type:   clusterv1.ClusterDeletingV1Beta2Condition,
+				Status: metav1.ConditionFalse,
+				Reason: clusterv1.ClusterDeletingDeletionTimestampNotSetV1Beta2Reason,
+			},
+		},
+		{
+			name:                    "deletionTimestamp set (some reason/message reported)",
+			cluster:                 fakeCluster("c", deleted(true)),
+			reconcileDeleteExecuted: true,
+			deletingReason:          clusterv1.ClusterDeletingWaitingForBeforeClusterDeleteHookV1Beta2Reason,
+			deletingMessage:         "Some message",
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.ClusterDeletingV1Beta2Condition,
+				Status:  metav1.ConditionTrue,
+				Reason:  clusterv1.ClusterDeletingWaitingForBeforeClusterDeleteHookV1Beta2Reason,
+				Message: "Some message",
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			setDeletingCondition(ctx, tc.cluster, tc.deletingReason, tc.deletingMessage)
+
+			deletingCondition := v1beta2conditions.Get(tc.cluster, clusterv1.ClusterDeletingV1Beta2Condition)
+			g.Expect(deletingCondition).ToNot(BeNil())
+			g.Expect(*deletingCondition).To(v1beta2conditions.MatchCondition(tc.expectCondition, v1beta2conditions.IgnoreLastTransitionTime(true)))
+		})
+	}
+}
+
+func TestSetAvailableCondition(t *testing.T) {
+	testCases := []struct {
+		name            string
+		cluster         *clusterv1.Cluster
+		expectCondition metav1.Condition
+	}{
+		{
+			name: "Takes into account all the required conditions",
+			cluster: &clusterv1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "machine-test",
+					Namespace: metav1.NamespaceDefault,
+				},
+				Spec: clusterv1.ClusterSpec{
+					Topology: nil, // not using CC
+				},
+				Status: clusterv1.ClusterStatus{
+					V1Beta2: &clusterv1.ClusterV1Beta2Status{
+						Conditions: []metav1.Condition{
+							// No condition reported yet, the required ones should be reported as missing.
+						},
+					},
+				},
+			},
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.ClusterAvailableV1Beta2Condition,
+				Status:  metav1.ConditionUnknown,
+				Reason:  v1beta2conditions.MultipleUnknownReportedReason,
+				Message: "InfrastructureReady: Condition InfrastructureReady not yet reported; ControlPlaneAvailable: Condition ControlPlaneAvailable not yet reported; WorkersAvailable: Condition WorkersAvailable not yet reported; RemoteConnectionProbe: Condition RemoteConnectionProbe not yet reported; Deleting: Condition Deleting not yet reported",
```

Comment:
This can be fixed in the summary util, but I will defer this to a follow up PR to avoid changing tests in other controllers (this PR is already too big)

Reply:
I'm getting used to PRs of this size :)